src/t8_cmesh/t8_cmesh_vertex_conn_tree_to_vertex.cxx

```diff
@@ -0,0 +1,166 @@
+ */
+
+#include <t8_cmesh.h>
+#include <t8_cmesh/t8_cmesh_types.h>
+#include <t8_cmesh/t8_cmesh_vertex_conn_tree_to_vertex.hxx>
+
+/* constructor from a given vertex to tree list. */
+t8_cmesh_vertex_conn_tree_to_vertex::t8_cmesh_vertex_conn_tree_to_vertex (
+  const t8_cmesh_t cmesh_from, t8_cmesh_t cmesh, const t8_cmesh_vertex_conn_vertex_to_tree &vtt)
+  : t8_cmesh_vertex_conn_tree_to_vertex ()
+{
+  T8_ASSERT (t8_cmesh_is_committed (cmesh_from));
+  T8_ASSERT (t8_cmesh_is_initialized (cmesh));
+  T8_ASSERT (vtt.is_committed ());
+
+  /* Since we need to collect all global ids of a tree before we can add the tree,
+   * we need a temporary buffer structure.
+   * Note that this is very memory intensive, doubling the required memory. */
+
+  const t8_locidx_t num_local_trees = t8_cmesh_get_num_local_trees (cmesh_from);
+  const t8_locidx_t num_ghosts = t8_cmesh_get_num_ghosts (cmesh_from);
+  const t8_locidx_t num_local_trees_and_ghosts = num_local_trees + num_ghosts;
+
+  /* Stores for each tree an array of local index and mapped global vertex.
+   * We need to store the local indices as well, since they will not be
+   * sorted. Hence we need to sort later. */
+  using global_local_index_pair = std::pair<t8_gloidx_t, t8_locidx_t>;
+  std::vector<std::vector<global_local_index_pair>> global_ids_per_tree (num_local_trees_and_ghosts);
+
+  /* Iterate over all entries of the vertex to tree list. */
+  for (auto &[global_vertex, vertex_list] : vtt) {
+    /* Iterate over all entries of the vertex list,
+     * thus, each entry is a local tree id and vertex index. */
+    for (auto &[tree_index, vertex_index] : vertex_list) {
+      global_local_index_pair new_pair (global_vertex, vertex_index);
+      global_ids_per_tree[tree_index].push_back (new_pair);
+    }
+  }
+  /* Now sort the list and add the global indices of each tree. */
+  for (t8_locidx_t itree = 0; itree < num_local_trees_and_ghosts; ++itree) {
+    auto &tree_indices = global_ids_per_tree[itree];
+    /* Sort according to local index. */
+    std::sort (tree_indices.begin (), tree_indices.end (),
+               [] (global_local_index_pair &a, global_local_index_pair &b) { return a.second < b.second; });
+
+    /* Compute number of vertices of this tree. */
+    /* Get the trees class depending on whether it is a local tree or ghost. */
+    const t8_eclass_t tree_class = itree < num_local_trees
+                                     ? t8_cmesh_get_tree_class (cmesh_from, itree)
+                                     : t8_cmesh_get_ghost_class (cmesh_from, itree - num_local_trees);
+
+    const int num_tree_vertices = t8_eclass_num_vertices[tree_class];
+
+    /* Size of list must match number of tree vertices. */
+    SC_CHECK_ABORTF (
+      tree_indices.size () == (size_t) num_tree_vertices,
+      "ERROR. Number of mapped local tree vertices for tree %i does not equal number of tree vertices. %zu != %i\n",
+      itree, tree_indices.size (), num_tree_vertices);
```

Comment:
It would then also shorten this to:
```suggestion
#if T8_ENABLE_DEBUG
    /* Check if no -1 is remaining in the nodes. If one is remaining, this node could not be mapped. */
    auto faulty_index = std::find(tree_indices.begin(), tree_indices.end(), -1);
    SC_CHECK_ABORTF (tree_indices.end() != faulty_index ,
      "ERROR. Node %i of tree %i could not be mapped to its global index.\n",
      std::distance(tree_indices.begin(), faulty_index), itree);
#endif /* T8_ENABLE_DEBUG */
```